index.js

```diff
@@ -91,26 +101,39 @@ async function init() {
 		const parsedUrl = new URL(query.get('url'));
 		[, user, repository, ref, dir] = urlParserRegex.exec(parsedUrl.pathname);
 
-		console.log('Source:', {user, repository, ref, dir});
+		isGHE = parsedUrl.hostname !== 'github.com';
+		if (isGHE) {
+			domain = parsedUrl.hostname;
+			apiEndpoint = `${parsedUrl.protocol}//${parsedUrl.host}/api/v3`;
+
+			document.querySelector('#create-token').href = `${parsedUrl.protocol}//${parsedUrl.host}/settings/tokens/new?description=Download GitHub directory&scopes=repo`;
+		}
+
+		console.log('Source:', {domain, user, repository, ref, dir});
```

Comment:
We don't necessarily need to log this. If it lets us reduce the parsing in this module, you can drop it

Reply:
I can drop this log but still, `domain`, `user`, `repo`, `ref` and `dir` are reused for `waitForToken` and `fetchPublicFile`.